bridge/slack/handlers.go

```diff
@@ -133,6 +133,15 @@ func (b *Bslack) skipMessageEvent(ev *slack.MessageEvent) bool {
 		ev.SubMessage.Edited == nil {
 		return true
 	}
+
+	for _, f := range ev.Files {
+		// If the file is in the cache and isn't older then a minute, skip it
+		if b.fileIsRecentlyCached(&f) {
+			b.Log.Debugf("Not downloading file id %s which we uploaded", f.ID)
+			return true
```

Comment:
It would be wrong to skip here. We should only skip (and thus return `true`) if **all** the files have been recently cached.

Reply:
but isn't this the filter for helping the bridge recognize it's own doing? it's not meant to be the place where the "make sure everything is cached" part of the process happens, is it? Or am I misunderstanding and the "was this us" logic for file_shares happens somewhere else?

Reply:
The code I would suggest instead:
```go
	var filesNotSkipped bool
	for _, f := range ev.Files {
		if !b.fileWasRecentlyCached(&f) {
			filesNotSkipped = true
		}
	}
	return !filesNotSkipped
```

Reply:
I don't find this very readable with all the inverted ifs and naming (lots of mental overhead) 
What if we make a separate function like below (maybe also rename `fileWasRecentlyCached` to `fileCached` maybe?)
```golang
func (b *Bslack) filesCached(files []slack.File) bool {
    for _, f:= range files {
        f := f
        if !b.fileWasRecentlyCached(&f) {
             return false
        }
   }
   return true
}
```
and just call
```golang
return filesCached(ev.Files)
```

Reply:
Sure! I like your suggestion. Indeeds less mental overhead. 👍

Reply:
I should caught this, but `return filesCached(ev.Files)` makes every message is skipped. Sorry, I went into no-think mode, without fully testing this last bit :)

I'm thinking this is how we should've ended this func:

```go
if len(ev.Files) > 0 {
    return b.filesCached(ev.Files)
}
return false
```